.github/workflows/build.yml

```
@@ -7,39 +7,55 @@ on:
   pull_request:
 
 jobs:
-  build_only:
-    name: Build Only
+  cleanup:
+    name: Cleanup
     runs-on: ubuntu-latest
+    steps:
+      - name: Remove unused Docker images
+        run: docker system prune -af
+      
+      - name: Clean up temporary files
+        run: sudo rm -rf /tmp/*
+        
+  setup:
+    name: Setup Buildx and Metadata
+    runs-on: ubuntu-latest
+    needs: cleanup
     steps:
       - uses: actions/checkout@v4
-
       - name: Set up Docker Buildx
         uses: docker/setup-buildx-action@v3
-
       - name: Docker meta
         id: meta
         uses: docker/metadata-action@v4
         with:
           images: pgc-images.sbgenomics.com/rokita-lab/openpedcanverse
           tags: |
             type=raw,value=analysisjob
-            # Only tag the image with latest if we're building on the default
-            # branch (e.g., main).
             type=raw,value=analysisjob,enable={{is_default_branch}}
+      - name: Upload meta outputs
+        uses: actions/upload-artifact@v4
+        with:
+          name: meta
+          path: meta.json
 
+build_only:
```

You're gonna laugh @jharenza... I think the issue here is spacing in the yaml... Yaml is very picky

```suggestion
  build_only:
```